ui/app/components/ui/identicon/blockieIdenticon/blockieIdenticon.component.js

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useRef, useState } from 'react'
+import PropTypes from 'prop-types'
+import { renderIcon } from '@download/blockies'
+
+const BlockieIdenticon = ({ address, diameter }) => {
+  const [dataUrl, setDataUrl] = useState(null)
+  const canvasRef = useRef(null)
+
+  useEffect(() => {
+    const canvas = canvasRef.current
+    renderIcon({ seed: address.toLowerCase() }, canvas)
+    const updatedDataUrl = canvas.toDataURL()
+
+    if (updatedDataUrl !== dataUrl) {
+      setDataUrl(updatedDataUrl)
+    }
+  })
+
+  return (
+    <>
+      <canvas
```

Comment:
There was a `createIcon` method that I could have used instead of `createIcon`, which would internally call `document.createElement('canvas')`. But I was concerned that that might open us up to a memory leak, so instead I'm using this canvas instance. The identicon is fully re-written each time it's drawn, and it doesn't allow transparent pixels, so we shouldn't need to worry about clearing it.